examples/threshold_signature/signature.rs

```diff
@@ -0,0 +1,49 @@
+use crate::threshold_signature::registration::ClosedRegistration;
+use blake2::digest::{Update, VariableOutput};
+use blake2::Blake2bVar;
+use blst::min_sig::{PublicKey, Signature};
+use blst::BLST_ERROR;
+
+/// Individual Signature.
+/// It includes a blst signature, its verification key, and the registration index of the signer.
+#[derive(Debug, Clone)]
+pub(crate) struct IndividualSignature {
+    pub(crate) signature: Signature,
+    pub(crate) verification_key: PublicKey,
+    pub(crate) signer_index: usize,
+}
+
+impl IndividualSignature {
```

Comment:
Not sure if it makes sense but prove/verify could be done in your signer (what I renamed "RegisteredSigner" in a comment) impl?